flink-job/src/main/java/com/ververica/field/dynamicrules/RulesEvaluator.java

```diff
@@ -158,22 +153,26 @@ public void run() throws Exception {
     return RulesSource.Type.valueOf(rulesSource.toUpperCase());
   }
 
-  private StreamExecutionEnvironment configureStreamExecutionEnvironment(
-      RulesSource.Type rulesSourceEnumType, boolean isLocal) {
+  private StreamExecutionEnvironment configureStreamExecutionEnvironment() {
+    final boolean isLocal = config.get(LOCAL_EXECUTION);
     Configuration flinkConfig = new Configuration();
-    flinkConfig.setBoolean(ConfigConstants.LOCAL_START_WEBSERVER, true);
 
-    StreamExecutionEnvironment env =
-        isLocal
-            ? StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(flinkConfig)
-            : StreamExecutionEnvironment.getExecutionEnvironment();
+    StreamExecutionEnvironment env;
+    if (isLocal) {
```

Comment:
Probably `isLocal` was not the best name for this option. This now mixes two things together a bit - the desire to run with a restart strategy that does not flood the logs if socket is not running and the desire not to start a WebUI in case you have something else running on the Flink UI port. If Flink will start the UI in a random, non occupied UI port and communicate this in the logs, I am for this change. If it causes failures, it makes sense to keep those parameters separate, but maybe better depict the intentions in their names.

Reply:
I read this `local` parameter as a shortcut for local execution without worrying too much about the details. I did not have the use case of something blocking your web UI port in my mind though.
In general, I would agree in separating these concerns, but I'm not sure we should follow that for this shortcut-parameter.
What do you think about this idea? Kust have a port(-range) as a parameter to `local` in the following way:

- `local` (without parameters): defaults, with web-UI on default port, fixedDelayRestart
- `local -1`: defaults, no web-UI, fixedDelayRestart
- `local <portOrPortRange>`: defaults, web-UI on some port in the given range, fixedDelayRestart - the user has to finde the port in the logs from this line:
`12:11:04.062 [main] INFO  o.a.f.r.d.DispatcherRestEndpoint - Web frontend listening at http://localhost:8081.`

Reply:
What would be the option to get non-fixedDelayRestart and no local WebUI? Omitting this parameter, like in the current version?

Reply:
FYI: finding this log line is currently not possible since too much is printed so that the output buffer is flushed too quickly (I addresses this in my performance improvements, so it will eventually be solved)

Reply:
Yes, I know, that is a problem. I have this setup right now:https://github.com/afedulov/state-migration/blob/master/src/main/resources/log4j2.yaml

BTW, I have started using yaml-based log4j configs and can recommend them - they are much easier to work with.

Reply:
yes, I would ignore non-fixedDelayRestart and no local WebUI - or actually, you can get this without setting `--local` in the first place (because then, nothing from that parameter is left)

Not sure setting Flink log level to error helps too much though as it also has some downsides during debugging / understanding Flink.

Reply:
> I would ignore non-fixedDelayRestart and no local WebUI  

Not sure I get what you men. I would like to keep this option for being able to run it on the platform, potentially for long time.

Reply:
If just not setting the local would get to no local UI and more resilient restarting strategy - this is perfectly fine.

Reply:
ok, let me summarise the options in the current state:
- no `local`: just plain `StreamExecutionEnvironment.getExecutionEnvironment();` with no other changes (taking server defaults)
- `local -1`: `StreamExecutionEnvironment.getExecutionEnvironment();` (with no web-UI) but also setting fixedDelayRestart
- `local <portOrPortRange>: `StreamExecutionEnvironment.createLocalEnvironmentWithWebUI()` with web-UI on some port in the given range, also setting fixedDelayRestart

I think, this covers all scenarios, most importantly not overriding the cluster defaults with no parameter (my original intention)